sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReactorReceiver.java

```diff
@@ -250,7 +250,7 @@ protected Message decodeDelivery(Delivery delivery) {
                 sink.error(new AmqpException(false, "updateDisposition failed while dispatching to Reactor.",
                     error, handler.getErrorContext(receiver)));
             }
-        });
+        }).cache().then(Mono.empty());  // cache because closeAsync use `when` to subscribe this Mono again.
```

Comment:
Do we need `.then(Mono.empty())`? If so, then use `then()`?

Reply:
Good catch. Mono.empty() is required here.